os/pm/pm_idle.c

```diff
@@ -164,13 +163,14 @@ void pm_idle(void)
 		/* Send signal to shutdown other cores here */
 		for (cpu = 1; cpu < CONFIG_SMP_NCPUS; cpu++) {
 			if (up_get_cpu_state(cpu) == CPU_RUNNING) {
+				/* Reset core gating status flag */
+				up_set_gating_flag_status(cpu, 0);
 				up_cpu_hotplug(cpu);
 			}
-			/* Reset core gating status flag */
-			up_set_gating_flag_status(cpu, 0);
 			/* Check whether each of the cpu has entered hotplug */
 			while(up_get_cpu_state(cpu) != CPU_HOTPLUG);
```

Comment:
can you please help to correct the code here? 
`while (`

Reply:
done